test/redis_locking_test.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+require_relative 'test_helper'
+require 'mocha/minitest'
+require 'minitest/stub_const'
+require 'ostruct'
+require 'sidekiq'
+require 'dynflow/executors/sidekiq/core'
+
+module Dynflow
+  module RedisLockingTest
+    describe Executors::Sidekiq::RedisLocking do
+      class Orchestrator
+        include Executors::Sidekiq::RedisLocking
+
+        attr_accessor :logger
+
+        def initialize(world, logger)
+          @world = world
+          @logger = logger
+        end
+      end
+
+      class Logger
+        attr_reader :logs
+        def initialize
+          @logs = []
+        end
+
+        [:info, :error, :fatal].each do |key|
+          define_method key do |message|
+            @logs << [key, message]
+          end
+        end
+      end
+
+      after do
+        ::Sidekiq.redis do |conn|
+          conn.del Executors::Sidekiq::RedisLocking::REDIS_LOCK_KEY
+        end
+      end
+
+      let(:world) { OpenStruct.new(:id => '12345') }
+      let(:orchestrator) { Orchestrator.new(world, Logger.new) }
+      let(:world2) { OpenStruct.new(:id => '67890') }
+      let(:orchestrator2) { Orchestrator.new(world2, Logger.new) }
+
+      it 'acquires the lock when it is not taken' do
+        orchestrator.wait_for_orchestrator_lock
+        logs = orchestrator.logger.logs
+        _(logs).must_equal [[:info, 'Acquired orchestrator lock, entering active mode.']]
```

Comment:
Do you think it would be possible to test the value in the redis directly (as we have the connection already, we could use it to assert the orchestrator behaviour, right?

Reply:
Good point, added.